### ISSUE TITLE
[docs] add stable release concept to docs infrastructure

### ISSUE THREAD

**Issue by schoudhury** [yugabyte/yugabyte-db#5402]
**Today with a single latest version that is used for both prod and non-prod**
v2.2 (latest version) - https://docs.yugabyte.com/latest/
v2.1 (earlier version) - https://docs.yugabyte.com/v2.1/
v2.0 (earlier version) - https://docs.yugabyte.com/v2.0/
v1.3 (earlier version) - https://docs.yugabyte.com/v1.3/
v1.2 (earlier version) - https://docs.yugabyte.com/v1.2/
v1.1 (earlier version) - https://docs.yugabyte.com/v1.1/
v1.0 (earlier version) - https://docs.yugabyte.com/v1.0/

**Required asap to support the new concept of latest (for non-prod) and stable (for prod) releases** 
v2.3 (latest version) - https://docs.yugabyte.com/latest/
v2.2 (stable version) - https://docs.yugabyte.com/stable/
v2.1 (earlier version) - https://docs.yugabyte.com/v2.1/
v2.0 (earlier version) - https://docs.yugabyte.com/v2.0/
v1.3 (earlier version) - https://docs.yugabyte.com/v1.3/

Additionally (nice to have)
-Delete all previous versions (as long as v1.0/v1.1/v1.2 links are auto redirected to https://docs.yugabyte.com/)

**Next major release**
v2.5 (latest version) - https://docs.yugabyte.com/latest/
v2.2 (stable version) - https://docs.yugabyte.com/stable/
v2.1 (earlier version) - https://docs.yugabyte.com/v2.1/
v2.0 (earlier version) - https://docs.yugabyte.com/v2.0/
v1.3 (earlier version) - https://docs.yugabyte.com/v1.3/

**1mo after Next major release**
v2.5 (latest version) - https://docs.yugabyte.com/latest/
v2.4 (stable version) - https://docs.yugabyte.com/stable/
v2.2 (earlier version) - https://docs.yugabyte.com/v2.2/
v2.1 (earlier version) - https://docs.yugabyte.com/v2.1/
v2.0 (earlier version) - https://docs.yugabyte.com/v2.0/
v1.3 (earlier version) - https://docs.yugabyte.com/v1.3/

cc @stevebang @rkarthik007

**Comment by yb-andrew** [comment]
This can be done by creating a stable directory and adding some extra logic in the version dropdown. The old links can be redirected to the home page with a Netlify.toml configuration.

**Comment by stevebang** [comment]
After we make this change, then I'll update the DocSearch configuration to match the changes.

**Comment by yb-andrew** [comment]
Committed new /stable content directory. Will now change Netlify redirect rules to remove references to all versions earlier than v1.3. Once that is complete and works correctly, I will delete the old directories.
